examples/01_icetray/03_deployment_to_i3.py

```diff
@@ -0,0 +1,127 @@
+from graphnet.training.loss_functions import LogCoshLoss
+
+logger = get_logger()
+
+# Constants
+features = FEATURES.UPGRADE
+truth = TRUTH.UPGRADE
+
+
+def construct_mock_model() -> StandardModel:
+    """Construct a mock model for the example.
+
+    Replace this with a trained model.
+    """
+    detector = IceCubeUpgrade(
+        graph_builder=KNNGraphBuilder(nb_nearest_neighbours=8),
+    )
+    gnn = DynEdge(
+        nb_inputs=detector.nb_outputs,
+        global_pooling_schemes=["min", "max", "mean", "sum"],
+    )
+    task = EnergyReconstruction(
+        hidden_size=gnn.nb_outputs,
+        target_labels=["energy"],
+        loss_function=LogCoshLoss(),
+        transform_prediction_and_target=torch.log10,
+    )
+    model = StandardModel(
+        detector=detector,
+        gnn=gnn,
+        tasks=[task],
+        optimizer_class=Adam,
+        optimizer_kwargs={"lr": 1e-03, "eps": 1e-03},
+    )
+    return model
+
+
+def construct_modules(
+    model_dict: Dict[str, Dict], gcd_file: str
+) -> List[I3InferenceModule]:
+    """Construct a list of I3InfereceModules for the I3Deployer."""
+    features = FEATURES.UPGRADE
+    deployment_modules = []
+    for model_name in model_dict.keys():
+        model_path = model_dict[model_name]["model_path"]
+        prediction_columns = model_dict[model_name]["prediction_columns"]
+        pulsemap = model_dict[model_name]["pulsemap"]
+        extractor = I3FeatureExtractorIceCubeUpgrade(pulsemap=pulsemap)
+        deployment_modules.append(
+            I3InferenceModule(
+                pulsemap=pulsemap,
+                features=features,
+                pulsemap_extractor=extractor,
+                model=model_path,
+                gcd_file=gcd_file,
+                prediction_columns=prediction_columns,
+                model_name=model_name,
+            )
+        )
+    return deployment_modules
+
+
+def main() -> None:
+    """GraphNeTI3Deployer Example."""
+    # configure input files, output folders and pulsemap
+    pulsemap = "SplitInIcePulses"
+    input_folders = [f"{TEST_DATA_DIR}/i3/upgrade_genie_step4_140028_000998"]
+    output_folder = f"{EXAMPLE_OUTPUT_DIR}/i3_deployment/upgrade"
+    gcd_file = f"{TEST_DATA_DIR}/i3/upgrade_genie_step4_140028_000998/GeoCalibDetectorStatus_ICUpgrade.v58.mixed.V0.i3.bz2"
+    input_files = []
+    for folder in input_folders:
+        input_files.extend(glob(join(folder, "*.i3.gz")))
+
+    # Configure Module dictionary & construct deployment modules
+    model_dict = {}
+    model_dict["graphnet_dynedge_energy_reconstruction"] = {
+        "model_path": construct_mock_model(),
+        "prediction_columns": ["energy_pred"],
+        "pulsemap": pulsemap,
+    }
+
+    deployment_modules = construct_modules(
+        model_dict=model_dict, gcd_file=gcd_file
+    )
+
+    # Construct I3 deployer
+    deployer = GraphNeTI3Deployer(
+        graphnet_modules=deployment_modules,
+        n_workers=1,
+        gcd_file=gcd_file,
+    )
```

Comment:
I think it could be beneficial to have two examples: One like this, that uses a `Deployer` class to parallelise I3 file processing; and one that just uses a simple I3Tray (the 20-or-so lines in the body of `GraphNeTI3Deployer._process_files`), i.e., what most people in IceCube are probably familier with.